### PR TITLE
Assign null instead of applying trailingHeaders#clear to clear traili…

### PR DESCRIPTION
**Pull request by imasahiro**
…ng headers correctly

### CODE REVIEW

**Comment by imasahiro** [comment]
I didn't reproduce on my laptop yet but according to our server log, it looks armeria try to break immutability when an error occurred.
Here is stacktrace for the error.
```
i.n.u.concurrent.AbstractEventExecutor   : A task raised an exception. Task: com.linecorp.armeria.common.stream.DefaultStreamMessage$$Lambda$118/1810829501@3a3b7674

java.lang.UnsupportedOperationException: immutable
    at com.linecorp.armeria.common.http.ImmutableHttpHeaders.unsupported(ImmutableHttpHeaders.java:587)
    at com.linecorp.armeria.common.http.ImmutableHttpHeaders.clear(ImmutableHttpHeaders.java:562)
    at com.linecorp.armeria.common.http.ImmutableHttpHeaders.clear(ImmutableHttpHeaders.java:29)
    at com.linecorp.armeria.common.http.HttpRequestAggregator.doClear(HttpRequestAggregator.java:49)
    at com.linecorp.armeria.common.http.HttpMessageAggregator.clear(HttpMessageAggregator.java:62)
    at com.linecorp.armeria.common.http.HttpRequestAggregator.onError(HttpRequestAggregator.java:43)
    at com.linecorp.armeria.common.stream.DefaultStreamMessage.notifySubscriberWithCloseEvent(DefaultStreamMessage.java:324)
    at com.linecorp.armeria.common.stream.DefaultStreamMessage.notifySubscriber(DefaultStreamMessage.java:259)
    at com.linecorp.armeria.common.stream.DefaultStreamMessage.lambda$notifySubscriber$1(DefaultStreamMessage.java:239)
    at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:163)
    at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:403)
    at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:304)
    at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:858)
    at io.netty.util.concurrent.DefaultThreadFactory$DefaultRunnableDecorator.run(DefaultThreadFactory.java:144)
    at java.lang.Thread.run(Thread.java:745)
```

File: core/src/main/java/com/linecorp/armeria/common/http/HttpRequestAggregator.java [hunk 1]
```diff
@@ -46,7 +46,7 @@ public void onError(Throwable t) {
 
     @Override
     protected void doClear() {
-        trailingHeaders.clear();
+        trailingHeaders = null;
```
Comment (trustin) [inline core/src/main/java/com/linecorp/armeria/common/http/HttpRequestAggregator.java]: How about assigning `HttpHeaders.EMPTY_HEADERS` instead?

Comment (imasahiro) [inline core/src/main/java/com/linecorp/armeria/common/http/HttpRequestAggregator.java]: I also thought about that and to think about consistency with HttpResponseAggregator, it was acceptable for me to assign null. (See [code](https://github.com/line/armeria/blob/f79467ad62eddd4b796fd25409cfdb63068831ac/core/src/main/java/com/linecorp/armeria/common/http/HttpResponseAggregator.java#L65))

Also, assigning HttpHeaders.EMPTY_HEADERS is also ok to me. So I fixed to use EMPTY_HEADERS.

Comment (trustin) [inline core/src/main/java/com/linecorp/armeria/common/http/HttpRequestAggregator.java]: Ah, if the consistency is the concern, yes, please assign it to null here or assign EMPTY_HEADERS there as well.

**Comment by trustin** [comment]
Thank you, @imasahiro !

**Pull request merged**